File: watermark-api/src/main/java/com/watermark/watermarkapi/domains/ImageUrl.java
package com.watermark.watermarkapi.domains;

public class ImageUrl {

	private final String url;
	private final String secureUrl;
	private final Integer userId;
	private final Integer pictureId;

	public ImageUrl(String url, String secureUrl, Integer userId, Integer pictureId) {
		this.url = url;
		this.secureUrl = secureUrl;
		this.userId = userId;
		this.pictureId = pictureId;
	}

	public Integer getPictureId() { return pictureId; }

	public Integer getUserId() { return userId; }

	public String getUrl() {
		return url;
	}

	public String getSecureUrl() {
		return secureUrl;
	}
}

File: watermark-client/src/app/domains/ImageUrl.ts
export class ImageUrl {
  url: string;
  secureUrl: string;
  userId: number;
  pictureId: number;

  constructor(url: string, secureUrl: string, userId: number, pictureId: number) {
    this.url = url;
    this.secureUrl = secureUrl;
    this.userId = userId;
    this.pictureId = pictureId;
  }

  public getUrl(): string {
    return this.url;
  }

  public getUserId(): number {
    return this.userId;
  }

  public getPictureId(): string {
    return this.url;
  }
}
